src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -389,52 +392,52 @@ internal static bool IsMinimalProgressRenderingEnabled()
 
             // if the activity is really long, only use up to half the width
             string activity;
-            if (Activity.Length > maxWidth / 2)
+            if (rawUI.LengthInBufferCells(Activity) > maxWidth / 2)
             {
-                activity = Activity.Substring(0, maxWidth / 2) + PSObjectHelper.Ellipsis;
+                activity = StringUtil.TruncateToBufferCellWidth(rawUI, Activity, maxWidth / 2) + PSObjectHelper.Ellipsis;
             }
             else
             {
                 activity = Activity;
             }
 
             // 4 is for the extra space and square brackets below and one extra space
-            int barWidth = maxWidth - activity.Length - indentation - 4;
+            int barWidth = maxWidth - rawUI.LengthInBufferCells(activity) - indentation - 4;
 
             var sb = new StringBuilder();
             int padding = maxWidth + PSStyle.Instance.Progress.Style.Length + PSStyle.Instance.Reverse.Length + PSStyle.Instance.ReverseOff.Length;
             sb.Append(PSStyle.Instance.Reverse);
 
             int maxStatusLength = barWidth - secRemainLength - 1;
-            if (maxStatusLength > 0 && StatusDescription.Length > barWidth - secRemainLength)
+            if (maxStatusLength > 0 && rawUI.LengthInBufferCells(StatusDescription) > barWidth - secRemainLength)
             {
-                sb.Append(StatusDescription.AsSpan(0, barWidth - secRemainLength - 1));
+                sb.Append(StringUtil.TruncateToBufferCellWidth(rawUI, StatusDescription, maxStatusLength));
                 sb.Append(PSObjectHelper.Ellipsis);
             }
             else
             {
                 sb.Append(StatusDescription);
             }
 
-            int emptyPadLength = barWidth + PSStyle.Instance.Reverse.Length - sb.Length - secRemainLength;
+            int emptyPadLength = barWidth - rawUI.LengthInBufferCells(sb.ToString()) - secRemainLength;
             if (emptyPadLength > 0)
             {
-                sb.Append(string.Empty.PadRight(emptyPadLength));
+                sb.Append(' ', emptyPadLength);
             }
 
             sb.Append(secRemain);
 
-            if (PercentComplete >= 0 && PercentComplete < 100 && barWidth > 0)
+            if (PercentComplete >= 0 && barWidth > 0)
```

Comment:
I don't think `PercentComplete < 100` should be removed. This check excludes the case where `PercentComplete == 100`.